mlir/lib/Dialect/Linalg/Transforms/Hoisting.cpp

```diff
@@ -43,6 +43,121 @@ using llvm::dbgs;
 using namespace mlir;
 using namespace mlir::linalg;
 
+scf::ForOp replaceWithDifferentYield(RewriterBase &rewriter, scf::ForOp loop,
+                                     Value newInitOperand, int index,
+                                     Value newYieldValue) {
+  OpBuilder::InsertionGuard g(rewriter);
+  rewriter.setInsertionPoint(loop.getOperation());
+  auto inits = llvm::to_vector(loop.getInits());
+
+  // Replace the init value with the new operand
+  inits[index] = newInitOperand;
```

Comment:
It looks wise to add a local assertion that `index` is in range. I know that `operator[]` asserts, but the message would be much less helpful.

Reply:
Added an assert  `assert(index < inits.size());`